File: src/main/java/net/pterodactylus/sone/notify/PostVisibilityFilter.java
package net.pterodactylus.sone.notify;

import java.util.function.Predicate;
import javax.annotation.Nonnull;
import javax.annotation.Nullable;

import net.pterodactylus.sone.data.Post;
import net.pterodactylus.sone.data.Sone;

import com.google.inject.ImplementedBy;

@ImplementedBy(DefaultPostVisibilityFilter.class)
public interface PostVisibilityFilter {

	boolean isPostVisible(@Nullable Sone sone, @Nonnull Post post);

	@Nonnull
	Predicate<Post> isVisible(@Nullable Sone currentSone);

}

File: src/main/kotlin/net/pterodactylus/sone/notify/PostVisibilityFilter.kt
package net.pterodactylus.sone.notify

import com.google.inject.ImplementedBy
import net.pterodactylus.sone.data.Post
import net.pterodactylus.sone.data.Sone

@ImplementedBy(DefaultPostVisibilityFilter::class)
@JvmDefaultWithoutCompatibility
interface PostVisibilityFilter {

	fun isPostVisible(sone: Sone?, post: Post): Boolean

	fun isVisible(currentSone: Sone?) =
			{ p: Post? -> p != null && isPostVisible(currentSone, p) }

}
